### PR TITLE
Service Object ADR

### PR DESCRIPTION
**Pull request by chrisrcoles**
## Description

Adds new ADRs for service objects. 

## Reviewer Notes

In-line comments

### CODE REVIEW

File: docs/adr/0033-service-object-layer.md [hunk 1]
```diff
@@ -0,0 +1,55 @@
+# *Service Object Layer*
+
+Currently the web service is built as two layers, Web Handlers ([pkg/handlers](https://github.com/transcom/mymove/tree/master/pkg/handlers)) which implement interfaces based on the swagger definitions of the services provided by the server and Model Objects ([pkg/models]((https://github.com/transcom/mymove/tree/master/pkg/models))) which marshal object representations of data in and out of the database.
```
Comment (chrisrcoles) [inline docs/adr/0033-service-object-layer.md]: Uses much of @ntwyman's [context from original ADR](https://github.com/transcom/mymove/blob/nt_dig_play/docs/adr/0031-service-layer-and-dependency-injection.md) on choosing a Dependency Injection Framework to give context.

File: docs/adr/0033-service-object-layer.md [hunk 2]
```diff
@@ -0,0 +1,55 @@
+# *Service Object Layer*
```
Comment (chrisrcoles) [inline docs/adr/0033-service-object-layer.md]: Focuses on decision to use service object layers.

File: .spelling [hunk 3]
```diff
@@ -428,4 +428,4 @@ _Regenerate
  - ./README.md
 transportation_service_provider_performances
 transportation_service_provider_performances
- - ./docs/data/tspp-data-creation.md
+ - ./docs/data/tspp-data-creation.md
```
Comment (ntwyman) [inline .spelling]: Looks like you killed a new line

File: docs/adr/0033-service-object-layer.md [hunk 4]
```diff
@@ -0,0 +1,55 @@
...
+
+Resources:
+
+* [Essential Ruby On Rails Patterns - Part 1: Service Objects](https://medium.com/selleo/essential-rubyonrails-patterns-part-1-service-objects-1af9f9573ca1)
+* [Using the Service Object Pattern in Go](https://www.calhoun.io/using-the-service-object-pattern-in-go/)
+* [Service Object Layer & Dependency Injection Design Document](https://docs.google.com/document/d/1xlqgVSTf9JUhZfWR18rvzaGPg2iHcF7uKRahGvrO45E/edit#)
+
+## Pros and Cons of the Alternatives
+
+### *Service Objects*
```
Comment (ntwyman) [inline docs/adr/0033-service-object-layer.md]: I'd like to get a little detail here about what flavor of Service Objects we are proposing. As our discussions have uncovered this means very different things to different people.  A very simplified example would be helpful here for illustrative if not prescriptive purposes.

File: docs/adr/0034-working-with-mocks-generation-and-assertion.md [hunk 5]
```diff
@@ -0,0 +1,61 @@
+# *Working With Mocks: Generation and Assertion*
```
Comment (ntwyman) [inline docs/adr/0034-working-with-mocks-generation-and-assertion.md]: I'd like to separate this out into a different PR.

I think the discussion of whether or not to use a Mocking framework and which framework to use is largely orthogonal.

Comment (mikena-truss) [inline docs/adr/0034-working-with-mocks-generation-and-assertion.md]: This says a lot about the capabilities of mocking frameworks, but I'm still missing *what this provides for us as developers and for our clients*.

We need concrete examples of why developers have needed mocking frameworks to get their job done. Was there a survey? Have developers mentioned this in the past?

We have a lot of folks who are still new to typed languages and learning how to use interfaces. Are we going to force them to write interfaces just to use a mocking library? Maybe we should focus on something like that instead and teach how interfaces can be beneficial to decouple dependencies in unit tests. If I never learned how to use a test implementation, I'd never understand the value of mocks.

We also need to see this in code. What's an example in the code base where this has caused an issue?

In the end, how does this transitively improve our clients experience? Are developers able to focus less on testing or build better tests which result in a better user experience? Do we spend less time debugging allowing us to build more and better features?

There are a lot technical details being identified here, but in the end, I just don't see the purpose.

File: docs/adr/0033-service-object-layer.md (hunk 4 above)
Comment (chrisrcoles) [reply docs/adr/0033-service-object-layer.md]: Ok, I've added another new [PR](https://github.com/transcom/mymove/pull/1742) that has service object documentation. Does adding a link to this new markdown file work?

File: docs/adr/0034-working-with-mocks-generation-and-assertion.md (hunk 5 above)
Comment (chrisrcoles) [reply docs/adr/0034-working-with-mocks-generation-and-assertion.md]: No problem - that's been done.

File: docs/adr/0033-service-object-layer.md (hunk 4 above)
Comment (ntwyman) [reply docs/adr/0033-service-object-layer.md]: 👍
